src/main/java/com/netgrif/application/engine/workflow/service/WorkflowService.java

```diff
@@ -151,12 +152,12 @@ public Case findOne(String caseId) {
 
     @Override
     public List<Case> findAllById(List<String> ids) {
-        List<Case> page = new LinkedList<>();
-        ids.forEach(id -> {
-            Optional<Case> useCase = repository.findById(id);
-            useCase.ifPresent(page::add);
-        });
-        if (page.size() > 0) {
+        List<Case> page = repository.findAllBy_idIn(ids).stream()
+                .filter(Objects::nonNull)
+                .sorted(Ordering.explicit(ids).onResultOf(Case::getStringId))
+                .collect(Collectors.toList());
+
+        if (!page.isEmpty()) {
```

Comment:
This should be part of stream processing now

Reply:
Done